### PR TITLE
feat: add source moderation count

### PR DESCRIPTION
**Pull request by ilasw**
Added `moderationPostCount` field to Source query only if _source member_ and _moderation is required_ .
Updated also tests including moderated squad to fixtures.

### CODE REVIEW

**Review by sshanzel — APPROVED** [review]
Looks good to me! 🚀

**Comment by sshanzel** [comment]
I think we can start merging this one now once we get the approval since the entity work has been merged.

File: __tests__/sources.ts
Comment (sshanzel) [inline __tests__/sources.ts:1126]: I think we are missing one more test. This is for when the user is not a moderator.

Comment (ilasw) [reply __tests__/sources.ts:1155]: Updated with more test cases ✅

File: src/schema/sources.ts
Comment (capJavert) [inline src/schema/sources.ts:1570]: This will need an index for sure over sourceId + createdById

Comment (ilasw) [reply src/schema/sources.ts:1570]: Added indexes!

File: src/migration/1730992663679-SourceModerationIndexes.ts
Comment (capJavert) [inline src/migration/1730992663679-SourceModerationIndexes.ts:12]: Missing drop index

File: src/schema/sources.ts
Comment (capJavert) [reply src/schema/sources.ts:1570]: We usually add the fields through graphorm, but since this is count I am fine with adding it here for now. Probably ideal to add this to flags as well, but let's see.

File: src/schema/sources.ts
Comment (capJavert) [inline src/schema/sources.ts:1576]: Realized just now what you were doing. 

We usually add the fields through graphorm, the difference is that if you request GQLSource anywhere graphorm will resolve the field, like this it only works on this specific query.

But since this is count I am fine with adding it here for now. Probably ideal to add this to flags as well so its not counted each time but let's not complicate for now since it was defined like this in RFC.

**Review by capJavert — APPROVED** [review]
Just missing down in mutations but good to merge after that.

**Pull request merged**
